### ISSUE TITLE
Show function signature while providing the arguments

### ISSUE THREAD

**Issue by mfronczyk** [haskell/haskell-language-server#3598]
**Is your feature request related to a problem? Please describe.**

When adding a function call, the extension suggests a list of functions with signatures, but when a function is selected and you start providing the parameters, the signature is not visible anymore and you need to bring the signature back sometimes for more complex functions.

**Describe the solution you'd like**

It'd be nice to keep the tooltip with the function signature open till you finish the call and make the current parameter bold there. It seems that the VS code extension for Idris has such feature, so maybe it'd be possible to add it here also?

Please see the Idris solution here - https://github.com/zjhmale/vscode-idris/commit/fd424291c73a03b61822603df65bf08bc945f00b

**Comment by alexnaspo** [comment]
I would be interested in working on this. I started hacking something together in ghcide but I'm wondering if this should be done within a HLS plugin?

**Comment by fendor** [comment]
It could be a plugin, but also part of ghcide directly, but let's see how the implementation looks like. If you require a slew of new dependencies, likely we want it in a plugin. If it basically can be done with the existing infrastructure, it is fine.

**Comment by michaelpj** [comment]
Also, sorry, forgot to link to the pre-existing issue that has some discussion: https://github.com/haskell/haskell-language-server/issues/2348

**Comment by michaelpj** [comment]
Hmm, possibly that is slightly different. I think that might be asking to show the argument doc in the main doc for an identifier? But we definitely also want them in signature help

**Comment by dan-blank** [comment]
I'd like to work on this ticket! :)

**Comment by michaelpj** [comment]
No, I think that issue _is_ different. Signature help is triggered by the user writing a character, i.e. while they're writing the function application. If we want argument haddock to be visible afterwards then we need it to be in the hover as well.

**Comment by samuel-williams-da** [comment]
@alexnaspo I see you have a WIP branch, are you still working on this?

**Comment by alexnaspo** [comment]
@samuel-williams-da I was having some trouble and got a bit busy so please don't let it hold up anyone else's work
